webapp/channels/src/utils/popouts/browser_popouts.ts

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useEffect} from 'react';
+import {useHistory} from 'react-router-dom';
+
+import {getHistory} from 'utils/browser_history';
+import {isDesktopApp} from 'utils/user_agent';
+
+const POPOUT_WIDTH = 800;
+const POPOUT_HEIGHT_OFFSET = 100;
+const CLOSE_POLL_INTERVAL = 200;
+const CLOSE_POLL_MAX_TRIES = 10;
+
+export function useBrowserPopout() {
```

Comment:
0/5 Unless it is only for internal use, I prefer 1 component per file, and hooks in a different file (preferably one hook per file).

But that is only my preference, so feel free to ignore :P

Reply:
I'm 0/5 on where it was goes. I might say to split it into its own file if it was a general purpose hook, but this is essentially internal code for popouts, so it's fine to organize it however you want. Like, if we were to pull out the popout stuff into its own package, I'd expect us to expose the `PopoutController` and some functions like `popoutThread`, but this wouldn't be exposed so it wouldn't matter where it was

Reply:
Moved to a separate file.